wl-sim/main/main.cpp

```diff
@@ -0,0 +1,125 @@
+#include <cstdlib>
+#include <time.h>
+
+#include "esp_log.h"
+#include "wl_sim.h"
+#include "random.h"
+#include "WL_Flash.h"
+
+static const char *TAG = "wl-sim";
+
+extern size_t restarted;
+extern size_t access_count;
+extern size_t move_count;
+extern uint32_t feistel_cycle_walks;
+
+// {constant,uniform,zipf,linear}
+#define ADDRESS_FUNCTION zipf
+// {0,1} if addr supplied to ADDRESS_FUNCTION should be zero
+#define ZERO_ADDR 0
+// {0,1} enable feistel network address randomization
+#define FEISTEL 1
+// number of iterations of main erase loop. BEWARE OF VERBOSE LOGGING
+#define ITERATIONS 25000000
+// {0,1} enable per sector verbose erase count logs
+#define VERBOSE_ERASE_COUNTS 0
+// {block_constant, block_zipf}
+#define BLOCK_SIZE_FUNCTION block_constant
+// block size of consecutive sectors erased per iteration
+#define ERASE_BLOCK 20
+// 0x1000 == 4K == full sector
+#define ERASE_SIZE (0x1000)
+
+// restart after each erase range, in per mille
+// 0 disables random restarting
+#define RESTART_PROBABILITY 0
+
+int main()
+{
+    srand(time(0));
+    esp_err_t result;
+
+    ESP_LOGI(TAG, "===== SETUP =====");
+    ESP_LOGI(TAG, "sector_size=0x%x (%u)", SECTOR_SIZE, SECTOR_SIZE);
+    ESP_LOGI(TAG, "sector_count=0x%x (%u)", SECTOR_COUNT, SECTOR_COUNT);
+
+#if FEISTEL == 1
+    init_feistel();
+#endif
+
+// test that feistel indeed maps 1:1, that no two sectors map to the same one
+#if 0
```

Comment:
Nitpicking: before finalising, please, remove #if 0 blocks and similar

Reply:
done